lib/grppccom.gi

```diff
@@ -1204,18 +1204,58 @@ end);
 
 #############################################################################
 ##
-#M  ComplementcClassesRepresentatives( <G>, <N> )
+#M  ComplementcClassesRepresentatives( <G>, <N> ) . . . . . . generic method
 ##
 InstallMethod( ComplementClassesRepresentatives,
-  "tell that the normal subgroup must be solvable",IsIdenticalObj,
-  [IsGroup,IsGroup],-2*RankFilter(IsGroup),
+  "by computing all subgroups", IsIdenticalObj, [ IsGroup, IsGroup ],
+  -2*RankFilter(IsGroup),
+
 function( G, N )
-  if IsSolvableGroup(N) then
+
+  local C, Hc, H;
+
+  # if <N> is trivial the only complement is <G>
+  if IsTrivial(N) then
+      C := [ G ];
+  # if <G> and <N> are equal the only complement is trivial
+  elif G = N  then
+      C := [ TrivialSubgroup(G) ];
+  elif not IsNormal(G, N) then
+    Error("N must be normal in G");
+  elif IsSolvableGroup(N) or HasSolvableFactorGroup(G, N) then
     TryNextMethod();
   fi;
-  Error("Cannot compute complement classes for nonsolvable normal subgroups");
+
+  Info(InfoWarning,1,"N and G/N are not solvable, computing all subgroups!");
+  C := [ ];
+  for Hc in ConjugacyClassesSubgroups(G) do
+    H := Representative(Hc);
+    if not IsTrivial(H) and G<>H and IsTrivial(Intersection(N, H)) and
```

Comment:
No, you are right. Probably should simply start with the size check if possible, and be done with it.